framework/src/systems/SystemBase.C

```diff
@@ -1070,6 +1087,9 @@ void
 SystemBase::update()
 {
   system().update();
+  std::vector<VariableName> std_field_variables;
+  getStandardFieldVariableNames(std_field_variables);
+  cacheVarIndicesByFace(std_field_variables);
```

Comment:
Do not understand why we need to do this in system update. System update usually means we `syn` solution

Reply:
We clear and rebuild all FaceInfo metadata from scratch whenever the mesh changes since all the face structure changes too.  Because of this, we need to rebuild all the var index caching too.